src/Illuminate/Filesystem/FilesystemAdapter.php

```diff
@@ -328,13 +333,29 @@ public function url($path)
      *
      * @param  \League\Flysystem\AwsS3v3\AwsS3Adapter  $adapter
      * @param  string  $path
+     * @param  int|string|\DateTime $expires The time at which the URL should
+     *                                      expire. This can be a Unix
+     *                                      timestamp, a PHP DateTime object,
+     *                                      or a string that can be evaluated
+     *                                      by strtotime().
      * @return string
      */
-    protected function getAwsUrl($adapter, $path)
+    protected function getAwsUrl($adapter, $path, $expires)
     {
-        return $adapter->getClient()->getObjectUrl(
-            $adapter->getBucket(), $adapter->getPathPrefix().$path
-        );
+        if ($expires === null) {
+            return $adapter->getClient()->getObjectUrl(
```

Comment:
Bro, I think this is the best practice you assign variables when you don't want to execute the same function twice but in this case, it's only executed once